### ISSUE TITLE
Maps view

### ISSUE THREAD

**Issue by bsramin** [metabase/metabase#8433]
Bug in "Region map" viewer.

- Browser and the version: All
- Databases: MySQL
- Metabase version: Latest (0.30.1)

If I try to generate a report with the "Region Map" with this GeoJson `https://cdn.rawgit.com/feries/geo-json/master/ITA_adm2.json` and using a query that generates me this output
```
1,Aosta
1,Catania
1,Lecce
1,Novara
1,Parma
1,Savona
1,Torino
1,Verona
```
(first column name: totale, second column name: nome)

generate this error in the javascript console log

![image](https://user-images.githubusercontent.com/672932/44917999-398c3b00-ad3a-11e8-8d91-506657778157.png)

and remain in infinite loading (and not loading the report correctly)

![image](https://user-images.githubusercontent.com/672932/44918026-4577fd00-ad3a-11e8-8971-bd05d0bad993.png)

**Comment by jornh** [comment]
Reproduced the bug! 

It seems this problem is more or less a variation (duplicate even?) of #7017

With:
* your https://cdn.rawgit.com/feries/geo-json/master/ITA_adm2.json as map source 
* Metabase 0.30.0
* All `1`'s or `42`'s it fails

Adjusting one or more of the values - so you get at least two color bins like this native query - or what I did in the picture below - and it works:

``` sql
VALUES ('Perugia', 42)
     , ('Cagliari', 42)
     , ('Verona', 42)
     , ('Bologna', 42)
     , ('Bari', 42)
     , ('Napoli', 42)
     , ('Grosetto', 42)
     , ('Torino', 42.0000000001)
```

![image](https://user-images.githubusercontent.com/766423/44925627-33548980-ad4f-11e8-9b29-1558cc956d95.png)

So in conclusion:
* Nothing wrong with your source GeoJSON - I guess this can also be reproduced with the built in US or world maps
* **Current workaround** - adjust your data to get more than one "color bin", doesn't matter if you use integers or floating point.

**Comment by bsramin** [comment]
Thanks! I hope someone can fix the bug 👍

**Comment by benoror** [comment]
Having a similar issue, I get the following exception in the browser console:

```
Uncaught TypeError: Cannot read property 'map' of undefined
    at t.value (app-main.bundle.js?78262faa1c529a733077:5)
    at u._renderValidatedComponentWithoutOwnerOrContext (vendor.bundle.js?78262faa1c529a733077:145)
    at u._renderValidatedComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at u.performInitialMount (vendor.bundle.js?78262faa1c529a733077:145)
    at u.mountComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at Object.mountComponent (vendor.bundle.js?78262faa1c529a733077:83)
    at J.mountChildren (vendor.bundle.js?78262faa1c529a733077:145)
    at J._createInitialChildren (vendor.bundle.js?78262faa1c529a733077:145)
    at J.mountComponent (vendor.bundle.js?78262faa1c529a733077:145)
    at Object.mountComponent (vendor.bundle.js?78262faa1c529a733077:83)
value @ app-main.bundle.js?78262faa1c529a733077:5
_renderValidatedComponentWithoutOwnerOrContext @ vendor.bundle.js?78262faa1c529a733077:145
_renderValidatedComponent @ vendor.bundle.js?78262faa1c529a733077:145
performInitialMount @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:83
mountChildren @ vendor.bundle.js?78262faa1c529a733077:145
_createInitialChildren @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:83
performInitialMount @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:83
mountChildren @ vendor.bundle.js?78262faa1c529a733077:145
_createInitialChildren @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:83
performInitialMount @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:145
mountComponent @ vendor.bundle.js?78262faa1c529a733077:83
updateChildren @ vendor.bundle.js?78262faa1c529a733077:145
_reconcilerUpdateChildren @ vendor.bundle.js?78262faa1c529a733077:145
_updateChildren @ vendor.bundle.js?78262faa1c529a733077:145
updateChildren @ vendor.bundle.js?78262faa1c529a733077:145
_updateDOMChildren @ vendor.bundle.js?78262faa1c529a733077:145
updateComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:83
_updateRenderedComponent @ vendor.bundle.js?78262faa1c529a733077:145
_performComponentUpdate @ vendor.bundle.js?78262faa1c529a733077:145
updateComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:83
_updateRenderedComponent @ vendor.bundle.js?78262faa1c529a733077:145
_performComponentUpdate @ vendor.bundle.js?78262faa1c529a733077:145
updateComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:83
_updateRenderedComponent @ vendor.bundle.js?78262faa1c529a733077:145
_performComponentUpdate @ vendor.bundle.js?78262faa1c529a733077:145
updateComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:145
receiveComponent @ vendor.bundle.js?78262faa1c529a733077:83
performUpdateIfNecessary @ vendor.bundle.js?78262faa1c529a733077:145
performUpdateIfNecessary @ vendor.bundle.js?78262faa1c529a733077:83
b @ vendor.bundle.js?78262faa1c529a733077:83
perform @ vendor.bundle.js?78262faa1c529a733077:94
perform @ vendor.bundle.js?78262faa1c529a733077:94
perform @ vendor.bundle.js?78262faa1c529a733077:83
w @ vendor.bundle.js?78262faa1c529a733077:83
close @ vendor.bundle.js?78262faa1c529a733077:83
closeAll @ vendor.bundle.js?78262faa1c529a733077:94
perform @ vendor.bundle.js?78262faa1c529a733077:94
perform @ vendor.bundle.js?78262faa1c529a733077:83
w @ vendor.bundle.js?78262faa1c529a733077:83
closeAll @ vendor.bundle.js?78262faa1c529a733077:94
perform @ vendor.bundle.js?78262faa1c529a733077:94
batchedUpdates @ vendor.bundle.js?78262faa1c529a733077:145
t @ vendor.bundle.js?78262faa1c529a733077:83
a @ vendor.bundle.js?78262faa1c529a733077:110
enqueueForceUpdate @ vendor.bundle.js?78262faa1c529a733077:110
s.forceUpdate @ vendor.bundle.js?78262faa1c529a733077:96
u @ vendor.bundle.js?78262faa1c529a733077:1
Show 33 more frames
```

**Comment by flamber** [comment]
I cannot reproduce this on 0.35.3 - think it was fixed in 0.32.9, if I remember correctly.
